src/wp-includes/block-template.php

```diff
@@ -238,13 +238,42 @@ function get_the_block_template_html() {
 	$content = $wp_embed->run_shortcode( $_wp_current_template_content );
 	$content = $wp_embed->autoembed( $content );
 	$content = do_blocks( $content );
+
+	// Write the global into a local variable so that it cannot be modified after this.
+	$parsed_post_content = $_wp_template_parsed_post_content;
+
+	// Replace all post content within the template with a temporary placeholder so that the post content is not
+	// unnecessarily processed again. See https://core.trac.wordpress.org/ticket/55996.
+	if ( is_array( $parsed_post_content ) ) {
+		foreach ( $parsed_post_content as $i => $post_content ) {
+			$pos = strpos( $content, $post_content );
+			if ( $pos !== false ) {
+				$content = substr_replace( $content, "<!-- wp-post-content-placeholder-$i -->", $pos, strlen( $post_content ) );
+			}
+		}
+	}
+
 	$content = wptexturize( $content );
 	$content = convert_smilies( $content );
 	$content = shortcode_unautop( $content );
-	$content = wp_filter_content_tags( $content );
+	$content = wp_filter_content_tags( $content, 'the_block_template' );
```

Comment:
This change is a bit unrelated, but since here this function is called outside of any filter, it is crucial to manually provide a `$context`. I went with `the_block_template` for now (similar to e.g. `the_content`), but certainly open to alternatives.

Reply:
Let's do this in another commit. No need to block this change as of this ticket.